spa/src/components/dashboard/sidebar/DashboardSidebarFooter.tsx

```
@@ -3,23 +3,76 @@ import { ICONS } from 'assets/icons/SvgIcon';
   const { user } = useUser();
   const { isOrgAdmin } = getUserRole(user);
   const [coreUpgradePromptClosed, setCoreUpgradePromptClosed] = useSessionState(SIDEBAR_CORE_UPGRADE_CLOSED, false);
-  const showCoreUpgradePrompt =
-    !coreUpgradePromptClosed &&
-    isOrgAdmin &&
-    user?.organizations[0].plan.name === 'FREE' &&
-    user?.revenue_programs[0].payment_provider_stripe_verified;
+  const [plusUpgradePromptClosed, setPlusUpgradePromptClosed] = useSessionState(SIDEBAR_PLUS_UPGRADE_CLOSED, false);
+  const [consultingUpgradePromptClosed, setConsultingUpgradePromptClosed] = useSessionState(
+    SIDEBAR_CONSULTING_UPGRADE_CLOSED,
+    false
+  );
+
+  const showUpgradePrompt = useMemo(() => {
+    // Doesn't show any prompts if plan is undefined
+    if (!user?.organizations[0]?.plan?.name) {
+      return false;
+    }
+
+    const isAdminAndStripeVerified = isOrgAdmin && user?.revenue_programs[0].payment_provider_stripe_verified;
+
+    if (user?.organizations[0].plan.name === PLAN_NAMES.FREE) {
+      return !coreUpgradePromptClosed && isAdminAndStripeVerified;
+    }
+
+    if (user?.organizations[0].plan.name === PLAN_NAMES.CORE) {
+      return !plusUpgradePromptClosed && isAdminAndStripeVerified;
+    }
+
+    if (user?.organizations[0].plan.name === PLAN_NAMES.PLUS) {
+      return !consultingUpgradePromptClosed && isAdminAndStripeVerified;
+    }
+
+    return false;
+  }, [
+    consultingUpgradePromptClosed,
+    coreUpgradePromptClosed,
+    isOrgAdmin,
+    plusUpgradePromptClosed,
+    user?.organizations,
+    user?.revenue_programs
+  ]);
+
+  console.log({ showUpgradePrompt, user });
+
+  const onClosePrompt = useCallback(() => {
+    if (user?.organizations[0].plan.name === PLAN_NAMES.PLUS) {
+      return setConsultingUpgradePromptClosed(true);
+    }
+
+    if (user?.organizations[0].plan.name === PLAN_NAMES.CORE) {
+      return setPlusUpgradePromptClosed(true);
+    }
+
+    return setCoreUpgradePromptClosed(true);
```

Instead of defaulting here, I'd prefer to either throw an error if `plan.name` doesn't match any of these, or do nothing and log a Sentry error. I'd lean toward throwing an error because we should never have shown the prompt in the first place.